packages/framework-provider-aws/src/library/searcher-adapter.ts

```diff
@@ -32,62 +31,150 @@ export async function searchReadModel(
+      }
+    })
+    .join(' AND ')
 }
 
-function placeholderBuilderFor(propName: string): (valueIndex: number) => string {
-  return (valueIndex: number) => `:${propName}_${valueIndex}`
+const attrCount: string[] = []
+function resetAttrCount(): void {
+  attrCount.length = 0
+}
+function placeholderBuilderFor(propName: string): (valueIndex: number, valueSubIndex?: number) => string {
+  return (valueIndex: number, valueSubIndex?: number) => {
+    const placeholder = `:${propName}_${valueIndex}` + (typeof valueSubIndex === 'number' ? `_${valueSubIndex}` : '')
+    if (attrCount.includes(placeholder)) return placeholderBuilderFor(propName)(valueIndex + 1)
+    attrCount.push(placeholder)
+    return placeholder
+  }
 }
 
-function buildExpressionAttributeNames(filters: Record<string, Filter<any>>): ExpressionAttributeNameMap {
+function buildExpressionAttributeNames(filters: FilterFor<any>): ExpressionAttributeNameMap {
   const attributeNames: ExpressionAttributeNameMap = {}
   for (const propName in filters) {
-    attributeNames[`#${propName}`] = propName
+    switch (propName) {
+      case 'not':
+        Object.assign(attributeNames, buildExpressionAttributeNames(filters[propName] as FilterFor<any>))
+        break
+      case 'and':
+      case 'or':
+        for (const filter of filters[propName] as Array<FilterFor<any>>) {
+          Object.assign(attributeNames, buildExpressionAttributeNames(filter as FilterFor<any>))
+        }
+        break
+      case 'includes':
+        break
+      default:
+        Object.entries(filters[propName] as FilterFor<any>).forEach(([prop, value]) => {
+          attributeNames[`#${propName}`] = propName
+          if (typeof value === 'object' && !Array.isArray(value)) {
+            Object.assign(attributeNames, buildExpressionAttributeNames({ [prop]: value }))
+          }
+        })
+        break
+    }
   }
   return attributeNames
 }
 
-function buildExpressionAttributeValues(filters: Record<string, Filter<any>>): ExpressionAttributeValueMap {
+function buildExpressionAttributeValues(filters: FilterFor<any>): ExpressionAttributeValueMap {
   const attributeValues: ExpressionAttributeValueMap = {}
-  for (const propName in filters) {
-    const filter = filters[propName]
-    const holder = placeholderBuilderFor(propName)
-    filter.values.forEach((value, index) => {
+  Object.entries(filters).forEach(([propName]) => {
+    switch (propName) {
+      case 'not':
+        Object.assign(attributeValues, buildExpressionAttributeValues(filters[propName] as FilterFor<any>))
+        break
+      case 'and':
+      case 'or':
+        for (const filter of filters[propName] as Array<FilterFor<any>>) {
+          Object.assign(attributeValues, buildExpressionAttributeValues(filter as FilterFor<any>))
+        }
+        break
+      default:
+        Object.assign(attributeValues, buildAttributeValue(propName, filters[propName]))
+        break
```

Comment:
This switch is pretty similar to the one in `buildExpressionAttributeNames`. It makes me wonder if we could abstract this somehow to make both functions simpler 🤔 (Not a requirement for this PR though)